pkg/docker/util.go

```diff
@@ -155,3 +159,85 @@ func parseRepositoryTag(repos string) (string, string, string) {
 	}
 	return repos, "", ""
 }
+
+// PullImage pulls the Docker image specifies by name taking the pull policy
+// into the account.
+// TODO: The 'force' option will be removed
+func PullImage(name string, d Docker, policy api.PullPolicy, force bool) (*PullResult, error) {
+	// TODO: Remove this after we deprecate --force-pull
+	if force {
+		policy = api.PullAlways
+	}
+
+	// When pull policy not specified, default to DefaultPullPolicy
+	if len(policy) == 0 {
+		policy = api.DefaultBuilderPullPolicy
```

Comment:
Yes, it is :-)

On Fri, Nov 6, 2015 at 4:00 PM, Ben Parees <notifications@github.com> wrote:

> In pkg/docker/util.go
> <https://github.com/openshift/source-to-image/pull/324#discussion_r44144586>
> :
>
> > @@ -155,3 +159,85 @@ func parseRepositoryTag(repos string) (string, string, string) {
> >  	}
> >  	return repos, "", ""
> >  }
> > +
> > +// PullImage pulls the Docker image specifies by name taking the pull policy
> > +// into the account.
> > +// TODO: The 'force' option will be removed
> > +func PullImage(name string, d Docker, policy api.PullPolicy, force bool) (*PullResult, error) {
> > +	// TODO: Remove this after we deprecate --force-pull
> > +	if force {
> > +		policy = api.PullAlways
> > +	}
> > +
> > +	// When pull policy not specified, default to DefaultPullPolicy
> > +	if len(policy) == 0 {
> > +		policy = api.DefaultBuilderPullPolicy
>
> @bparees <https://github.com/bparees> the other place this is used is to
> pull previous image that has the policy explicitly set.
> I think we should not default to any policy and just return error when the
> policy is not set.
>
> isn't that what i suggested? :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/source-to-image/pull/324/files#r44144586>.
>



-- 
-----------------------------------------------
Michal Fojtik <mi@mifo.sk <mfojtik@mifo.sk>>
Red Hat OpenShift, Engineering